examples/httpbin-demo/src/charm.py

```diff
@@ -32,43 +31,44 @@ class HttpbinDemoCharm(ops.CharmBase):
 
     def __init__(self, framework: ops.Framework):
         super().__init__(framework)
+        framework.observe(self.on.collect_unit_status, self._on_collect_status)
         framework.observe(self.on['httpbin'].pebble_ready, self._on_httpbin_pebble_ready)
         framework.observe(self.on.config_changed, self._on_config_changed)
+        self.log_level = str(self.config['log-level'])  # str() is here for the type checker.
+        self.container = self.unit.get_container('httpbin')
+
+    def _on_collect_status(self, event: ops.CollectStatusEvent):
+        """Report the status of the workload (runs after each event)."""
+        if self.log_level.lower() not in VALID_LOG_LEVELS:
+            event.add_status(ops.BlockedStatus(f"invalid log level: '{self.log_level}'"))
+            return
+        if not self.container.can_connect():
+            event.add_status(ops.MaintenanceStatus('waiting for container'))
+            return
+        try:
+            if not self.container.get_service('httpbin').is_running():
+                event.add_status(ops.MaintenanceStatus('waiting for workload'))
+                return
+        except ops.ModelError:
+            event.add_status(ops.MaintenanceStatus('waiting for Pebble'))
```

Comment:
Nit: it feels like the Juju user probably shouldn't need to know a lot about Pebble (although once they start to debug they certainly would) so maybe "waiting for the workload container" would be more the terms they'd expect?